### PR TITLE
Support ASG update policies

### PR DESCRIPTION
**Pull request by simonirwin-r7**
Includes unit tests.

These attributes are configurable:
```
   update_policy do
          pause_time 'PT10M'
          min_instances_in_service 1
          max_batch_size 2
   end
```

Example converge where the AMI for the LaunchConfiguration has been changed
```
    converge  Stack nginxserver
update_in_progress  sirwin-test-staging-nginxserver: (AWS::CloudFormation::Stack/arn:aws:cloudformation:us-east-1:XXXXXX:stack/sirwin-test-staging-nginxserver/063470e0-ffe2-11e5-8971-500c217dbefe) User Initiated
update_in_progress  NginxServerLaunchConfig: (AWS::AutoScaling::LaunchConfiguration/sirwin-test-staging-nginxserver-NginxServerLaunchConfig-1M84UGZVHUV2O) Requested update requires the creation of a new physical resource; hence creating one.
update_in_progress  NginxServerLaunchConfig: (AWS::AutoScaling::LaunchConfiguration/sirwin-test-staging-nginxserver-NginxServerLaunchConfig-15M3I6QD5KH64) Resource creation Initiated
update_complete  NginxServerLaunchConfig: (AWS::AutoScaling::LaunchConfiguration/sirwin-test-staging-nginxserver-NginxServerLaunchConfig-15M3I6QD5KH64) 
update_in_progress  NginxServerAutoScalingGroup: (AWS::AutoScaling::AutoScalingGroup/sirwin-test-staging-nginxserver-NginxServerAutoScalingGroup-1VZ7SJSD3JI0T) 
update_in_progress  NginxServerAutoScalingGroup: (AWS::AutoScaling::AutoScalingGroup/sirwin-test-staging-nginxserver-NginxServerAutoScalingGroup-1VZ7SJSD3JI0T) Rolling update initiated. Terminating 1 obsolete instance(s) in batches of 1, while keeping at least 1 instance(s) in service. Pausing for PT5M when new instances are added to the autoscaling group.
update_in_progress  NginxServerAutoScalingGroup: (AWS::AutoScaling::AutoScalingGroup/sirwin-test-staging-nginxserver-NginxServerAutoScalingGroup-1VZ7SJSD3JI0T) Temporarily setting autoscaling group MinSize and DesiredCapacity to 2.
update_in_progress  NginxServerAutoScalingGroup: (AWS::AutoScaling::AutoScalingGroup/sirwin-test-staging-nginxserver-NginxServerAutoScalingGroup-1VZ7SJSD3JI0T) New instance(s) added to autoscaling group - Pausing for PT5M.
update_in_progress  NginxServerAutoScalingGroup: (AWS::AutoScaling::AutoScalingGroup/sirwin-test-staging-nginxserver-NginxServerAutoScalingGroup-1VZ7SJSD3JI0T) Terminating instance(s) [i-862dd601]; replacing with 0 new instance(s).
update_in_progress  NginxServerAutoScalingGroup: (AWS::AutoScaling::AutoScalingGroup/sirwin-test-staging-nginxserver-NginxServerAutoScalingGroup-1VZ7SJSD3JI0T) Successfully terminated instance(s) [i-862dd601] (Progress 100%).
update_complete  NginxServerAutoScalingGroup: (AWS::AutoScaling::AutoScalingGroup/sirwin-test-staging-nginxserver-NginxServerAutoScalingGroup-1VZ7SJSD3JI0T) 
update_complete_cleanup_in_progress  sirwin-test-staging-nginxserver: (AWS::CloudFormation::Stack/arn:aws:cloudformation:us-east-1:XXXXXXXX:stack/sirwin-test-staging-nginxserver/063470e0-ffe2-11e5-8971-500c217dbefe) 
delete_in_progress  NginxServerLaunchConfig: (AWS::AutoScaling::LaunchConfiguration/sirwin-test-staging-nginxserver-NginxServerLaunchConfig-1M84UGZVHUV2O) 
delete_complete  NginxServerLaunchConfig: (AWS::AutoScaling::LaunchConfiguration/sirwin-test-staging-nginxserver-NginxServerLaunchConfig-1M84UGZVHUV2O) 
update_complete  sirwin-test-staging-nginxserver: (AWS::CloudFormation::Stack/arn:aws:cloudformation:us-east-1:XXXXXXX:stack/sirwin-test-staging-nginxserver/063470e0-ffe2-11e5-8971-500c217dbefe)
```

### CODE REVIEW

File: lib/convection/model/template/resource/aws_auto_scaling_auto_scaling_group.rb [hunk 1]
```diff
@@ -30,6 +30,51 @@ class AutoScalingGroup < Resource
...
+
+          def update_policy(&block)
+            @update_policy = UpdatePolicy.new
+            @update_policy.instance_exec(&block)
+          end
+
+          ##
+          # UpdatePolicy for AWS::AutoScaling::AutoScalingGroup
+          ##
+          class UpdatePolicy
```
Comment (athompson-r7) [inline lib/convection/model/template/resource/aws_auto_scaling_auto_scaling_group.rb]: Can this be split out like the resource property types?

Comment (simonirwin-r7) [inline lib/convection/model/template/resource/aws_auto_scaling_auto_scaling_group.rb]: @athompson-r7 I'll try that out now

Comment (athompson-r7) [inline lib/convection/model/template/resource/aws_auto_scaling_auto_scaling_group.rb]: Might what to create a separate dir for this, it is a resource attribute not a resource property.  I think the pattern still applies though.  See this [PR](https://github.com/rapid7/convection/pull/46) for an example.

Comment (simonirwin-r7) [inline lib/convection/model/template/resource/aws_auto_scaling_auto_scaling_group.rb]: The top-level resource.rb file currently handles the other supported resource attributes (DependsOn, DeletionPolicy).  I'll try to separate out UpdatePolicy; I'm loathe to change the other attributes if there are no tests in place.

**Comment by simonirwin-r7** [comment]
This needs some more work around rendering

**Pull request closed without merging**